### ISSUE TITLE
Link to module without doc is rendered to hex doc

### ISSUE THREAD

**Issue by chulkilee** [elixir-lang/ex_doc#1014]
I have an umbrella project with apps and extra markdown file (passing it in `extras` option). 

In the extra markdown file, I referred `Db.Repo.init/2`, `Db.Post` and `Ecto.Repo`. And I put `@moduledoc false` to `Db.Repo` module.

Results are:

- correct: `Ecto.Repo` => a link to hexdocs page with version (probably in the mix.lock)
- correct: `Db.Post` => a link to `Db.Package.html` 
- **incorrect**: `Db.Repo.init/2` is rendered as a link to hexdocs page with `db` package with umbrella app version - i.e. https://hexdocs.pm/db/0.0.1/Db.Repo.html#init/2

Environment:

- ex_doc: 0.20.2
- Elixir: 1.8.2

Expected result:

Do not make a link if module/func has doc.

**Comment by josevalim** [comment]
There is open pull request to do this for the same project but links across
projects will always assume a certain structure and it would otherwise be
very expensive to check.

I know this is a very simplistic answer but can you not link to something
that does not exist publicly?
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by josevalim** [comment]
Another option is to allow dependencies to be excluded, so we never link to
them. Maybe there is already an API for this but it is also annoying to
have to specify things you don’t want to link.
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by chulkilee** [comment]
Oh, I found #951 and #1009 are for this. Closing this issue.

I'm just mentioning module/func with ` - and actually I was surprised (in a good way) when ex_doc adds links even for extra markdown file (not in elixir code) (e.g. `mix test` is also got link!)

 I'd like to keep it as `Db.Repo` in my markdown since

- it's also rendered by other markdown supporting system (e.g. github/gitlab/..),
- ex_doc does auto link - so it should care about this, not requiring other changes in markdown to "fix" auto link issue... *ideally*.
